documentation/scaling_puppet_server.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: default
+title: "Scaling Puppet Server with compile masters"
+canonical: "/puppetserver/latest/scaling_puppet_server.html"
+---
+
+To scale Puppet Server for many thousands of nodes, you'll need to add Puppet masters dedicated to catalog compilation. These Servers are known as **compile masters**, and are simply additional load-balanced Puppet Servers that receive catalog requests from agents and synchronize the results with each other.
+
+## Planning your load-balancing strategy
+
+The rest of your configuration depends on how you plan on distributing the agent load. Determine what your deployment will look like before you add any compile masters, but **implement load balancing as the last step** only after you have the infrastructure in place to support it.
+
```

Comment:
Might be worth calling out the PE docs on load balancing:

https://docs.puppet.com/pe/latest/install_lei_load.html

The stuff on DNS and SRV records looks great, but PE users will need to take the above documentation into consideration as well.